onnxruntime/core/providers/cpu/reduction/reduction_ops.cc

```diff
@@ -822,10 +844,57 @@ static void ValidateKeepDims(const Tensor* input, int64_t keepdims) {
   ValidateKeepDims(input->Shape(), keepdims);
 }
 
+template <typename AGG>
+bool check_and_reduce_empty_set_input(OpKernelContext* ctx, const gsl::span<const int64_t> axes, bool keepdims) {
+  const Tensor* input = ctx->Input<Tensor>(0);
+  const TensorShape& input_shape = input->Shape();
+  if (input_shape.Size() != 0) {
+    return false;
+  }
+
+  // input is an empty set
+  std::vector<int64_t> input_axes;
```

Comment:
nit: instead of copying to another vector `input_axes`, I wonder if you can use a span. e.g., either use `axes` directly or assign `axes_tensor->DataAsSpan<int64_t>()` to it.

Reply:
great suggestion @edgchen1 ! My concern modifying axis as an input may cause unexpected issue for the caller. Direct use of axes_tensor->DataAsSpan<int64_t>() may hurt readability. Thank you for the suggestion anyway.

Reply:
it's not modifying the input `axes` as that is a `span<const int64_t>`.